Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -282,6 +283,78 @@ public ActionResult ReadIpsContaining(string value)
             return this.Json(ips, JsonRequestBehavior.AllowGet);
         }
 
+        public void MoveSubmission(int categoryId)
+        {
+            ContestCategory category = this.Data.ContestCategories.GetById(categoryId);
+
+            foreach (var categoryContest in category.Contests)
+            {
+                var competeOnlyParticipants = categoryContest
+                .Participants
+                .GroupBy(p => p.UserId)
+                .Where(g => g.Count() == 1 && g.All(p => p.IsOfficial))
+                .SelectMany(gr => gr.ToList());
+
+                foreach (Participant participant in competeOnlyParticipants)
+                {
+                    foreach (var participantScore in participant.Scores)
+                    {
+                        participantScore.IsOfficial = false;
+                    }
+
+                    participant.IsOfficial = false;
+                }
+
+                var competeAndPracticeParticipants = categoryContest
+                    .Participants
+                    .GroupBy(p => p.UserId)
+                    .Where(g => g.Count() == 2)
+                    .ToDictionary(grp => grp.Key, grp => grp.OrderBy(p => p.IsOfficial).ToList());
+
+                foreach (KeyValuePair<string, List<Participant>> competeAndPracticeParticipant in competeAndPracticeParticipants)
+                {
+                    var unofficialParticipant = competeAndPracticeParticipants[competeAndPracticeParticipant.Key][0];
+                    var officialParticipant = competeAndPracticeParticipants[competeAndPracticeParticipant.Key][1];
```

Comment:
IMO it's better value of the dictionary to be `IEnumerable<Participant>` and to get `First()` And `Last()`. I think it's more readable.